src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,292 @@
+package scala
+package collection.mutable
```

Comment:
@DarkDimius : So, this rewrite will happen for 2.13 or a further future version of Scala? I would like to get this into Scala sooner than later (if we have to wait for a full rewrite of collections). Should I send this pull request against Scala then?